crates/composer/src/jukebox.rs

```diff
@@ -0,0 +1,60 @@
+use eyre::{Context, Result};
+use rodio::{
+    source::{Buffered, SamplesConverter},
+    Decoder, OutputStreamHandle, Source,
+};
+use std::{fs::File, io::BufReader, path::Path};
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+pub(crate) enum Sample {
+    Click,
+}
+
+impl Sample {
+    const ALL: &[Sample] = &[Self::Click];
+
+    fn filename(&self) -> &'static Path {
+        Path::new(match self {
+            Sample::Click => "click.wav",
+        })
+    }
+}
+
+type Buffer = Buffered<SamplesConverter<Decoder<BufReader<File>>, f32>>;
+
+/// Records (samples) are loaded to the jukebox once, and then in can quickly play any of them.
+pub struct Jukebox {
+    samples: Vec<(Sample, Buffer)>,
```

Comment:
Any reason not to use `HashMap<Sample, Buffer>`? I can't think of a reason to have multiple `Buffer`s for a `Sample`. It would also simplify and speed up* the lookup in `play`.

* with negligible effect, if any at all

Reply:
Yeah the `Vec` is I think a premature optimization of mine. I think I've read somewhere that for a "small" number of elements (which was still surprisingly high for modern processors), linearly traversing a vector was actually faster than asymptotically better structures like a hash map (because it is more CPU cache-friendly).

But this is extremely premature (and doing optimization without a benchmark is a sin of its own), so I've replaced that with the `HashMap` in fixup commit.